src/annotator/anchoring/pdf.js

```diff
@@ -448,55 +480,51 @@ export function anchor(root, selectors) {
  * @param {HTMLElement} root - The root element
  * @param {Range} range
  */
-export function describe(root, range) {
-  const normalizedRange = new BrowserRange(range).normalize();
+export async function describe(root, range) {
+  // "Shrink" the range so that the start and endpoints are at offsets within
+  // text nodes rather than any containing nodes.
+  try {
+    range = TextRange.fromRange(range).toRange();
+  } catch {
+    throw new Error('Selection does not contain text');
+  }
 
-  const startTextLayer = getNodeTextLayer(normalizedRange.start);
-  const endTextLayer = getNodeTextLayer(normalizedRange.end);
+  const startTextLayer = getNodeTextLayer(range.startContainer);
+  const endTextLayer = getNodeTextLayer(range.endContainer);
 
-  if (startTextLayer !== endTextLayer) {
-    return Promise.reject(
-      new Error('selecting across page breaks is not supported')
-    );
+  if (!startTextLayer || !endTextLayer) {
+    throw new Error('Selection is outside page text');
   }
 
-  const startRange = normalizedRange.limit(startTextLayer);
-  const endRange = normalizedRange.limit(endTextLayer);
-
-  if (!startRange || !endRange) {
-    return Promise.reject(new Error('range is outside text layer'));
+  if (startTextLayer !== endTextLayer) {
+    throw new Error('Selecting across page breaks is not supported');
   }
 
-  const startPageIndex = getSiblingIndex(startTextLayer.parentNode);
+  const startPos = TextPosition.fromPoint(
+    range.startContainer,
+    range.startOffset
+  ).relativeTo(startTextLayer);
+
+  const endPos = TextPosition.fromPoint(
+    range.endContainer,
+    range.endOffset
+  ).relativeTo(endTextLayer);
```

Comment:
Pedantically, this is equivalent to `startTextLayer`, right? Is it doing any disservice here to refer to `endTextLayer`? I mean, the symmetry is nice as-is.

Reply:
You're right. At present you could replace `endTextLayer` with `startTextLayer` here and it would work just the same. I figure though that `endTextLayer` is what you'd expect to see. It's also one less thing to change if we support selecting across page breaks in future.

Reply:
Yes, for readability, this version is better.